libs/src/sdk/services/Auth/types.ts

```diff
@@ -0,0 +1,23 @@
+import type { EIP712TypedData, MessageData } from 'eth-sig-util'
+
+export type TransactionData = MessageData<EIP712TypedData>['data']
+
+export type AuthService = {
+  /**
+   * Get a shared secret, used for Chats
+   */
+  getSharedSecret: (publicKey: string | Uint8Array) => Promise<Uint8Array>
+  /**
+   * Sign some data, used for Chats
+   */
+  sign: (data: string) => Promise<[Uint8Array, number]>
+  /**
+   * Sign an ethereum transaction, used for EntityManager writes
+   * NOTE: audius-client can stub out until we use sdk for writes in the client
+   */
+  signTransaction: (data: TransactionData) => Promise<string>
```

Comment:
Still keeping this separate because the types are different, maybe we could make the types align?

Reply:
I think it makes sense to keep them separate since they are two different things requiring two different fns, one is for signing arbitrary messages and one is for structured data

Reply:
@sliptype actually just thinking about this again, i guess we could infer what to do based on the type of the data that's passed in (string or obj)? 
and either always return a UIntArray or a string

Reply:
Interesting, that is a fair amount of responsibility for one function but might make the interface cleaner. I'll give it a try and see how it ends up looking

Reply:
yeah honestly i think either way is fine

Reply:
but sg!!